onadata/libs/serializers/metadata_serializer.py

```diff
@@ -101,23 +96,91 @@ def create(self, validated_data):
             from_kpi=from_kpi,
         )
 
-    def _validate_data_file_type(self, data_file_type, data_file, data_value):
+    def _validate_data_value(self, attrs):
+        if self.instance and self.instance.pk:
+            # ToDo , Should we allow PATCHing the object with different file?
+            attrs['data_file'] = attrs.get(
+                'data_file', self.instance.data_file
+            )
+            attrs['data_type'] = attrs.get(
+                'data_type', self.instance.data_type
+            )
+            try:
+                attrs['data_value']
+            except KeyError:
+                attrs['data_value'] = self.instance.data_value
+                return attrs
 
-        data_value = (
-            data_file.name if data_file else data_value
-        )
-        allowed_types = settings.SUPPORTED_MEDIA_UPLOAD_TYPES
+        data_value = attrs.get('data_value')
+
+        if attrs.get('data_type') == 'media' and attrs.get('data_file') is None:
+            URLValidator(message=_("Invalid url %s." % data_value))(data_value)
```

Comment:
I guess this could safely be:
```python
data_value = attrs['data_value']

if attrs['data_type'] == 'media' and attrs['data_file'] is None:
    # better style for python3?
    URLValidator(message=_("Invalid url {}.").format(data_value))(data_value)
```

Reply:
Yep, let's use `f-strings`